squidpy/pl/_spatial_utils.py

```diff
@@ -0,0 +1,875 @@
+    scalebar_units: Sequence[str] | None = None,
+    scalebar_kwargs: Mapping[str, Any] = MappingProxyType({}),
+) -> Axes:
+
+    ax.set_yticks([])
+    ax.set_xticks([])
+
+    ax.set_xlabel(fig_params.ax_labels[0])
+    ax.set_ylabel(fig_params.ax_labels[1])
+
+    ax.autoscale_view()
+
+    if value_to_plot is None:
+        # if only dots were plotted without an associated value
+        # there is not need to plot a legend or a colorbar
+        return ax
+
+    if legend_fontoutline is not None:
+        path_effect = [patheffects.withStroke(linewidth=legend_fontoutline, foreground="w")]
+    else:
+        path_effect = []
+
+    # Adding legends
+    if is_categorical_dtype(color_source_vector):
+        clusters = color_source_vector.categories
+        _palette = _get_palette(adata, cluster_key=value_to_plot, categories=clusters, palette=palette)
+        _add_categorical_legend(
+            ax,
+            color_source_vector,
+            palette=_palette,
+            scatter_array=coords,
+            legend_loc=legend_loc,
+            legend_fontweight=legend_fontweight,
+            legend_fontsize=legend_fontsize,
+            legend_fontoutline=path_effect,
+            na_color=na_color,
+            na_in_legend=na_in_legend,
+            multi_panel=True if fig_params.axs is not None else False,
+        )
+    else:
+        # TODO: na_in_legend should have some effect here
+        plt.colorbar(cax, ax=ax, pad=0.01, fraction=0.08, aspect=30)
+
+    cur_coords = np.concatenate([ax.get_xlim(), ax.get_ylim()])
+
+    if img is not None:
+        ax.imshow(img, cmap=img_cmap, alpha=img_alpha)
+    else:
+        ax.set_aspect("equal")
+        ax.invert_yaxis()
+
+    if crops is not None:
+        ax.set_xlim(crops.to_tuple()[0], crops.to_tuple()[1])
+        ax.set_ylim(crops.to_tuple()[3], crops.to_tuple()[2])
+    else:
+        ax.set_xlim(cur_coords[0], cur_coords[1])
+        ax.set_ylim(cur_coords[3], cur_coords[2])
+
+    if isinstance(scalebar_dx, list) and isinstance(scalebar_units, list):
+        scalebar = ScaleBar(scalebar_dx[lib_count], units=scalebar_units[lib_count], **scalebar_kwargs)
+        ax.add_artist(scalebar)
+
+    return ax
+
+
+def _map_color_seg(
+    seg: NDArrayA,
+    cell_id: NDArrayA,
+    color_vector: NDArrayA | pd.Series[CategoricalDtype],
+    color_source_vector: pd.Series[CategoricalDtype],
+    cmap_params: CmapParams,
+    seg_erosionpx: int | None = None,
+    seg_boundaries: bool = False,
+    seg_transparent: bool = False,
+) -> NDArrayA:
+
+    cell_id = np.array(cell_id)
+    if is_categorical_dtype(color_vector):
+        if any(color_source_vector.isna()) and seg_transparent:
+            cell_id[color_source_vector.isna()] = 0
+        val_im: NDArrayA = map_array(seg, cell_id, color_vector.codes + 1)  # type: ignore
+        cols = colors.to_rgba_array(color_vector.categories)  # type: ignore
+    else:
+        val_im = map_array(seg, cell_id, cell_id)
```

Comment:
this is needed to "filter out" non-existings segmentation masks and then replace with color in the `label2rgb` call